src/js/tabs/motors.js

```diff
@@ -365,8 +365,28 @@ TABS.motors.initialize = function (callback) {
         // select current mixer configuration
         mixerListElement.val(FC.MIXER_CONFIG.mixer).change();
 
+        function validateMixerOutputs() {
+            MSP.promise(MSPCodes.MSP_MOTOR).then(() => {
+                const motors = mixerList[FC.MIXER_CONFIG.mixer - 1].motors;
+                self.numberOfValidOutputs = motors;
```

Comment:
Why are we setting this?

Reply:
`i < motors` falls within scope. If 0 motors found in FC_MOTOR_DATA[0] === 0 and the mixer model has more than 0 motors defined we know we have not enough resources and reset the mixer to default.

371 looks duplicate to line 376, but in the event the loop wouldn't evaluate there is no initialization.
I have tested with F4 (having 6 outputs) and F7 (having 8 outputs) with all models.

Reply:
Not sure I understand - how will this work to reset the model if say we select a hexacopter mixer (`motors === 6`) but only have 4 motor outputs defined (`i === 4`)?

Reply:
The first check on line 375 determines the number of outputs, hence the return statement on line 380
So in this case `i` would stop at 4 and because motors > 0 we fire reset.

Reply:
Have rebased to make it more readable, but something is wrong with some items.
Something has changed because this was definitely working before:

The problem observed are this items:
- `Heli90`, `Custom Airplane` and `Custom Tricopter` returns now 0 output.
-  `Singlecopter` changes into `Custom`
- `PPM to servo` changes into `Custom Airplane`

Reply:
@mikeller The issue is also with BFC 10.7 and FW 4.0.4 so not sure how I could miss this. Made the dialog somewhat more informative. Going to create an issue for this on firmware.

![Screenshot from 2021-03-24 16-05-16](https://user-images.githubusercontent.com/8344830/112333319-c7deb880-8cba-11eb-92a8-b773dbdd8fcd.png)

Reply:
Changed dialog:

![Screenshot from 2021-03-24 21-33-37](https://user-images.githubusercontent.com/8344830/112382780-91ba2c80-8cec-11eb-9744-26650bb0d27c.png)

Reply:
Updated the dialog to reflect the need for custom mmix:
![Screenshot from 2021-03-24 22-07-02](https://user-images.githubusercontent.com/8344830/112383381-579d5a80-8ced-11eb-9c1c-b95d9259f72f.png)